lib/vehicle_factory.rb

```diff
@@ -0,0 +1,105 @@
+class VehicleFactory
+  attr_reader :vehicles
+
+  def initialize
+    @vehicles = []
+  end
+
+  def create_vehicles(data_service)
+    data_service.each do |vehicle_info|
+      formatted_vehicle_info = {
+        vin: change_to_vin_key(vehicle_info),
+        year: vehicle_info[:model_year],
+        make: vehicle_info[:make],
+        model: change_to_model_key(vehicle_info),
+        engine: change_to_engine_key(vehicle_info),
+        county: vehicle_info[:county]
+        }
+        @vehicles << Vehicle.new(formatted_vehicle_info)
+    end
+  end
+
+  def change_to_engine_key(vehicle_info)
+    if vehicle_info.has_key?(:electric_vehicle_type)
+      :ev
+    elsif vehicle_info[:fuel_type] == "GAS"
```

Comment:
Line 23 uses the has_key? method but here you use the bracket notation. Either one is fine, but use the same one for consistency.